src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -5862,6 +5862,15 @@ bool Unit::Attack(Unit* victim, bool meleeAttack)
         if (victim->GetTypeId() == TYPEID_PLAYER)
             victim->SetInCombatWith(this);
 
+        // also enter combat with pet/guardian's owner if exists
+        if (Unit* owner = victim->GetOwner())
```

Comment:
Other way around. If a creature attacks your pet/guardian, you will be in combat.